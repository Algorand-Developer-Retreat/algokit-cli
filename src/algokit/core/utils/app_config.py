from pathlib import Path

import click

AppName = "algokit"


def get_config_dir():
    return Path(click.get_app_dir(AppName))


def read(file_name: str):
    return get_config_dir().joinpath(file_name).read_text("utf-8")


def write(file_name: str, content: str):
    get_config_dir().mkdir(parents=True, exist_ok=True)
    config_file = get_config_dir().joinpath(file_name)
    click.echo(click.style(f"Writing content to {config_file}", fg="blue"))
    config_file.write_text(content, "utf-8")
