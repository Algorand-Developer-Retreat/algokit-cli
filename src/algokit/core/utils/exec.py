import subprocess
from pathlib import Path

import click


def run(
    command: str,
    working_directory: Path | None = None,
    suppress_output: bool = False,
    throw_on_error: str | None = None,
):
    click.echo(click.style(f"Running '{command}' in '{working_directory or Path.cwd()}'", fg="blue"))
    result = subprocess.run(command.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=working_directory)
    if not suppress_output and result.stdout.decode("utf-8").strip().__len__() > 0:
        click.echo(click.style(result.stdout.decode("utf-8"), fg="blue"))

    if throw_on_error is not None and result.returncode != 0:
        click.echo(click.style(result.stderr.decode("utf-8"), fg="red"))
        raise click.ClickException(throw_on_error)

    return result
