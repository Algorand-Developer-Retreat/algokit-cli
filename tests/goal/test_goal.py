from approvaltests import verify  # type: ignore
from utils.app_dir_mock import AppDirs
from utils.click_invoker import invoke
from utils.exec_mock import ExecMock


def test_goal_no_args(app_dir_mock: AppDirs, exec_mock: ExecMock):
    result = invoke("goal")

    assert result.exit_code == 0
    verify(result.output)


def test_goal_simple_args(app_dir_mock: AppDirs, exec_mock: ExecMock):
    result = invoke("goal account list")

    assert result.exit_code == 0
    verify(result.output)


def test_goal_complex_args(app_dir_mock: AppDirs, exec_mock: ExecMock):
    result = invoke("goal account export -a RKTAZY2ZLKUJBHDVVA3KKHEDK7PRVGIGOZAUUIZBNK2OEP6KQGEXKKUYUY")

    assert result.exit_code == 0
    verify(result.output)


def test_goal_start_without_docker(app_dir_mock: AppDirs, exec_mock: ExecMock):
    exec_mock.should_fail_on("docker version")

    result = invoke("goal")

    assert result.exit_code == 1
    verify(result.output)


def test_goal_start_without_docker_engine_running(app_dir_mock: AppDirs, exec_mock: ExecMock):
    exec_mock.should_bad_exit_on("docker version")

    result = invoke("goal")

    assert result.exit_code == 1
    verify(result.output)
