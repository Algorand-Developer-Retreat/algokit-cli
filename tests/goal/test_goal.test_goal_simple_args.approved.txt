DEBUG: Running 'docker version' in '{current_working_directory}'
DEBUG: docker: STDOUT
DEBUG: docker: STDERR
DEBUG: Running 'docker exec algokit_algod goal account list' in '{current_working_directory}'
 STDOUT
 STDERR
